### DISCUSSION TITLE
pg_dump error due to insufficient permission of postgres role

### DISCUSSION THREAD

**Discussion by d-mok** [supabase/supabase#19914] [Questions]
When I run pg_dump with the role `postgres`, I get this error:
```
pg_dump: error: query failed: ERROR:  permission denied for table schema_migrations
pg_dump: detail: Query was: LOCK TABLE auth.users, auth.refresh_tokens, auth.instances, auth.audit_log_entries, auth.schema_migrations, storage.buckets, storage.objects, storage.migrations, public.question, public.folder, public.teacher, public.source, auth.identities, realtime.schema_migrations, realtime.subscription, auth.sessions, auth.mfa_factors, auth.mfa_challenges, auth.mfa_amr_claims, auth.sso_providers, auth.sso_domains, auth.saml_providers, auth.saml_relay_states, auth.flow_state IN ACCESS SHARE MODE
ERROR: Could not complete database dump - please check full error log (EX_USAGE).
```
`postgres` role does not have any permission on the table `realtime.schema_migrations`. In fact, in the supabase dashboard, I can't even see the table `realtime.schema_migrations` and `realtime.subscription`. It seems that the `postgres` role has no permission on anything in the `realtime` schema.
I tried to:
`grant all on realtime.schema_migrations TO postgres`
but get the error:
`ERROR:  42501: permission denied for table schema_migrations`.
And there is no way to upgrade `postgres` to `superuser` now.
Is there anything I can do? I have recently upgrade the database to postgresql 15. Not sure if this has anything to do with the situation.

**Comment by johannbuscail** [comment]
Hey I'm facing exactly the same issue. Have you found a solution ?

**Reply by d-mok** [reply]
i added the flag `--schema=public` to pg_dump without backing up the protected schemas.

**Reply by johannbuscail** [reply]
Oh ok. Was this while trying to integrate prisma in your app ? If so, did it work in the end ?
